astropy/coordinates/angles.py

```diff
@@ -650,27 +672,11 @@ def __setitem__(self, item, value):
         super().__setitem__(item, value)
         self._wrap_internal()
 
-    def _wrap_internal(self):
+    def _wrap_internal(self, wrap_angle=None):
         """
-        Wrap the internal values in the Longitude object. Using the
-        :meth:`~astropy.coordinates.Angle.wrap_at` method causes
-        recursion.
+        Wrap the internal values in the Longitude object.
         """
-        # Convert the wrap angle and 360 degrees to the native unit of
-        # this Angle, then do all the math on raw Numpy arrays rather
-        # than Quantity objects for speed.
-        a360 = u.degree.to(self.unit, 360.0)
-        wrap_angle = self.wrap_angle.to_value(self.unit)
-        wrap_angle_floor = wrap_angle - a360
-        self_angle = self.value
-        # Do the wrapping, but only if any angles need to be wrapped
-        #
-        # This invalid catch block can be removed when the minimum numpy
-        # version is >= 1.19 (NUMPY_LT_1_19)
-        with np.errstate(invalid='ignore'):
-            wraps = (self_angle - wrap_angle_floor) // a360
-        if np.any(wraps != 0):
-            self -= (wraps * a360) << self.unit
+        self._wrap_at(self.wrap_angle)
```

Comment:
Is this right? there's an added arg ``wrap_angle=None``, but it never gets used.

Reply:
Oops, that's a mistake! I started with an override of the method, but then decided against it. Will remove that addition.

Reply:
Actually, on second thought, let me use remove `_wrap_internal` altogether...